### PR TITLE
Adding SetDomain API in OperatorSchema to allow operator registeration for different domains.

### PR DESCRIPTION
**Pull request by linkerzhang**

### CODE REVIEW

**Comment by gramalingam** [comment]
So, I guess this is just the API exposed for registering ops with a domain? I guess the registry implementation will also have to be extended (just like in Edward's PR 11 for version-number)? Is the plan to do that separately? Thanks.

**Comment by linkerzhang** [comment]
@gramalingam , Thank you! I'll do the operator registry implementation change. I assume that you're talking about the Schema function in OpSchemaRegistry (say, calling with a op_type, version, domain, an op schema will be returned. I'll wait for Edward's change checked in firstly, otherwise, I'll have to resolve conflicts, you know. :)

File: onnx/defs/gen_doc.py [hunk 1]
```diff
@@ -46,20 +46,22 @@ def main(args):
...
+          if domain:
+              domain = ' (Domain: {})'.format(domain)
+          s = '* <a href="#{}">{}{}</a>{}\n'.format(
+              op_type, support_level_str(schema.support_level), op_type, domain)
           fout.write(s)
 
       fout.write('\n')
 
-      for _, op_type, schema in sorted_ops:
+      for domain, _, op_type, schema in sorted_ops:
```
Comment (bddppq) [inline onnx/defs/gen_doc.py]: nit: since you don't use `domain` in this block, you can do `for _, _, op_type, schema ...`

**Comment by bddppq** [comment]
LG

**Pull request merged**